cmd/watch_repo/add.go

```diff
@@ -0,0 +1,309 @@
+					}
+				}
+			}
+		case "k", "up":
+			switch a.view {
+			case addItemType:
+				if a.itemType > 0 {
+					a.itemType--
+					return a, nil
+				}
+			case addDroneServer:
+				for i, server := range droneServers {
+					if server == a.server {
+						if i > 0 {
+							a.server = droneServers[i-1]
+							return a, nil
+						}
+					}
+				}
+			}
+		case "esc":
+			return a, exitAddItemCmd()
+		case "enter":
+			switch a.view {
+			case addItemType:
+				switch a.itemType {
+				case githubPrItem, githubReleaseItem:
+					a.view = addItemRepo
+					return a, tea.Batch(a.itemRepo.Focus(), textinput.Blink)
+				case droneBuildItem:
+					a.server = droneRancherPrServer
+					a.view = addDroneServer
+				}
+			case addDroneServer:
+				a.view = addItemRepo
+				return a, tea.Batch(a.itemRepo.Focus(), textinput.Blink)
+			case addItemRepo:
+				a.view = addItemId
+				return a, tea.Batch(a.itemID.Focus(), textinput.Blink)
+			case addItemId:
+				return a, a.validateItem()
+			}
+			return a, nil
+		}
+	case error:
+		fmt.Println("got an error", msg)
+		return a, nil
+	}
+
+	if a.view == addItemRepo {
+		var cmd tea.Cmd
+		a.itemRepo, cmd = a.itemRepo.Update(msg)
+		return a, cmd
+	}
+	if a.view == addItemId {
+		var cmd tea.Cmd
+		a.itemID, cmd = a.itemID.Update(msg)
+		return a, cmd
+	}
+
+	return a, nil
+}
+
+func (a addItem) View() string {
+	switch a.view {
+	case addItemType:
+		return a.chooseTypeView()
+	case addDroneServer:
+		return a.chooseDroneView()
+	case addItemRepo:
+		return a.repoInputView()
+	case addItemId:
+		return a.idInputView()
+	}
+
+	return ""
+
+}
+
+func (a addItem) chooseTypeView() string {
+	t := a.itemType
+	tpl := "What kind of resource do you want to watch?\n"
+	tpl += "\n" + checkbox("Drone Build", t == droneBuildItem)
+	tpl += "\n" + checkbox("GitHub Pull Request", t == githubPrItem)
+	tpl += "\n" + checkbox("GitHub Release", t == githubReleaseItem)
+	tpl += "\n\n"
+	tpl += subtle.Render("j/k, up/down: select")
+	tpl += dot + subtle.Render("enter: choose")
+	tpl += dot + subtle.Render("q, esc: quit")
+
+	return tpl
+}
+
+func (a addItem) chooseDroneView() string {
+	tpl := "Choose Drone server:\n"
+
+	for _, server := range droneServers {
+		tpl += "\n"
+		switch server {
+		case droneRancherPrServer:
+			tpl += checkbox("drone-pr.rancher.io", server == a.server)
+		case droneRancherPubServer:
+			tpl += checkbox("drone-publish.rancher.io", server == a.server)
+		case droneK3sPrServer:
+			tpl += checkbox("drone-pr.k3s.io", server == a.server)
+		case droneK3sPubServer:
+			tpl += checkbox("drone-publish.k3s.io", server == a.server)
+		}
+	}
+	tpl += "\n\n"
+	tpl += subtle.Render("j/k, up/down: select")
+	tpl += dot + subtle.Render("enter: choose")
+	tpl += dot + subtle.Render("q, esc: quit")
+	return tpl
+}
+
+func (a addItem) repoInputView() string {
+	tpl := "What is the repo?\n\n"
+	tpl += a.itemRepo.View()
+	tpl += "\n\n" + subtle.Render("enter: save") + dot + subtle.Render("q, esc: quit") + "\n"
```

Comment:
`q` won't quit, since it is an input

```suggestion
	tpl += "\n\n" + subtle.Render("enter: save") + dot + subtle.Render("esc: quit") + "\n"
```